deployer/troubleshooting-integration/src/main/java/io/hyscale/troubleshooting/integration/actions/FixCrashingApplication.java

```diff
@@ -15,55 +15,68 @@
+	private static final Logger logger = LoggerFactory.getLogger(FixCrashingApplication.class);
 
-    @Override
-    public void process(TroubleshootingContext context) {
-        Object obj = context.getAttribute(FailedResourceKey.FAILED_POD);
-        String lastState = null;
-        if (obj != null) {
-            V1Pod pod = (V1Pod) FailedResourceKey.FAILED_POD.getKlazz().cast(obj);
-            lastState = pod != null ? PodStatusUtil.lastStateOf(pod) : null;
-        }
+	@Override
+	public void process(TroubleshootingContext context) {
+		Object obj = context.getAttribute(FailedResourceKey.FAILED_POD);
+		String lastState = null;
+		V1Pod pod = null;
+		Integer statusCode = null;
+		if (obj != null) {
+			pod = (V1Pod) FailedResourceKey.FAILED_POD.getKlazz().cast(obj);
+			lastState = pod != null ? PodStatusUtil.lastStateOf(pod) : null;
+		}
 
-        DiagnosisReport report = new DiagnosisReport();
-        if (lastState != null) {
-            if (lastState.equals(PodStatus.OOMKILLED.getStatus())) {
-                report.setReason(AbstractedErrorMessage.NOT_ENOUGH_MEMORY_FOUND.formatReason(context.getServiceInfo().getServiceName()));
-                report.setRecommendedFix(AbstractedErrorMessage.NOT_ENOUGH_MEMORY_FOUND.getMessage());
-            } else if (lastState.equals(PodStatus.COMPLETED.getStatus())) {
-                report.setReason(AbstractedErrorMessage.INVALID_STARTCOMMANDS_FOUND.getReason());
-                report.setRecommendedFix(AbstractedErrorMessage.INVALID_STARTCOMMANDS_FOUND.getMessage());
-            } else {
-                report.setReason(AbstractedErrorMessage.APPLICATION_CRASH.getReason());
-                report.setRecommendedFix(AbstractedErrorMessage.APPLICATION_CRASH.getMessage());
-            }
-        } else {
-            report.setReason(AbstractedErrorMessage.APPLICATION_CRASH.getReason());
-            report.setRecommendedFix(AbstractedErrorMessage.APPLICATION_CRASH.getMessage());
-        }
-        context.addReport(report);
-    }
+		DiagnosisReport report = new DiagnosisReport();
+		if (lastState != null) {
+			if (lastState.equals(PodStatus.OOMKILLED.getStatus())) {
+				report.setReason(AbstractedErrorMessage.NOT_ENOUGH_MEMORY_FOUND
+						.formatReason(context.getServiceInfo().getServiceName()));
+				report.setRecommendedFix(AbstractedErrorMessage.NOT_ENOUGH_MEMORY_FOUND.getMessage());
+			} else if (lastState.equals(PodStatus.COMPLETED.getStatus())) {
+				report.setReason(AbstractedErrorMessage.INVALID_STARTCOMMANDS_FOUND.getReason());
+				report.setRecommendedFix(AbstractedErrorMessage.INVALID_STARTCOMMANDS_FOUND.getMessage());
+			} else {
+				V1ContainerState v1ContainerState = PodStatusUtil.getLastState(pod);
+				statusCode = PodStatusUtil.getExitCode(v1ContainerState);
+				PodStatusCode.Signals singanls = PodStatusCode.Signals.fromCode(statusCode);
+				if (singanls != null) {
+					if (PodStatusCode.Signals.statusCodeVsMessage.get(singanls) != null)
```

Comment:
Map key is Integer right, it wont contain signals
Should we use PodStatusCode.Signals.statusCodeVsMessage.contains(signals.getCode())